### PR TITLE
CORTX-31968 Jenkins pipeline to create build, deploy it and execute performance sanity

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
-  Custom-ci with performance sanity support - This job will enable the developer to run the performance sanity using manual Jenkins execution on feature branches. This job will perform the below tasks,

   - Generate CORTX container images from provided feature branch details
   - Deploy CORTX with generated images. 
   - Execute performance sanity on the deployed cluster. 

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Performance/job/performance-check/7/console 

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/performance/run-performace-tests-functions.sh [hunk 1]
```diff
@@ -92,7 +95,8 @@ function update_setup_confiuration() {
 function execute_perfpro() {
     yum install ansible -y
     pushd $SCRIPT_LOCATION/performance/PerfPro
-    ansible-playbook perfpro.yml -i inventories/hosts --extra-vars '{ "EXECUTION_TYPE" : "sanity" ,"REPOSITORY":{"motr":"cortx-motr","rgw":"cortx-rgw"} , "COMMIT_ID": { "main" : "d1234c" , "dev" : "a5678b"},"PR_ID" : "cortx-rgw/1234" , "USER":"Shailesh Vaidya","GID" : "729494" }' -v
+    add_primary_separator "Executing Ansible CLI"
+    ANSIBLE_LOG_PATH=$ANIBLE_LOG_FILE ansible-playbook perfpro.yml -i inventories/hosts --extra-vars '{ "EXECUTION_TYPE" : "sanity" ,"REPOSITORY":{"motr":"cortx-motr","rgw":"cortx-rgw"} , "COMMIT_ID": { "main" : "d1234c" , "dev" : "a5678b"},"PR_ID" : "cortx-rgw/1234" , "USER":"Shailesh Vaidya","GID" : "729494" }' -v
```
Comment (gauravchaudhari02) [inline scripts/performance/run-performace-tests-functions.sh:99]: Are these vars going to be hardcoded?

File: jenkins/performance-ci/performance-check.groovy [hunk 2]
```diff
@@ -0,0 +1,140 @@
+/* groovylint-disable DuplicateStringLiteral, LineLength, NestedBlockDepth */
```
Comment (gauravchaudhari02) [inline jenkins/performance-ci/performance-check.groovy:1]: should we name this file as `performance-ci.groovy` as it is ci for performance?

File: jenkins/performance-ci/nightly-performance-ci.groovy [hunk 3]
```diff
@@ -0,0 +1,122 @@
+pipeline {
+    agent {
+        node {
+            label 'docker-centos-7.9.2009-node'
```
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply jenkins/performance-ci/nightly-performance-ci.groovy:4]: We are using this node only as a jump host so OS does not matter.

File: jenkins/performance-ci/nightly-performance-ci.groovy [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        disableConcurrentBuilds()
+    }
+    parameters {
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Cluster Setup scripts', trim: true)
+        string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repository for Cluster Setup scripts', trim: true)
+        string(name: 'CORTX_SERVER_IMAGE', defaultValue: 'ghcr.io/seagate/cortx-rgw:2.0.0-latest', description: 'CORTX-RGW image', trim: true)
+        string(name: 'CORTX_DATA_IMAGE', defaultValue: 'ghcr.io/seagate/cortx-data:2.0.0-latest', description: 'CORTX-DATA image', trim: true)
+        string(name: 'CORTX_CONTROL_IMAGE', defaultValue: 'ghcr.io/seagate/cortx-control:2.0.0-latest', description: 'CORTX-CONTROL image', trim: true)
+        string(name: 'CORTX_TOOLS_BRANCH', defaultValue: 'sanity-fix', description: 'Repository for Cluster Setup scripts', trim: true)
+        string(name: 'CORTX_TOOLS_REPO', defaultValue: 'shailesh-vaidya/seagate-tools', description: 'Repository for Cluster Setup scripts', trim: true)
```
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply jenkins/performance-ci/nightly-performance-ci.groovy:23]: Nop. It's only still the Performance team add changes. Refer - https://jts.seagate.com/browse/CORTX-30154?focusedCommentId=2379049&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2379049

File: jenkins/performance-ci/performance-check.groovy (hunk 2 above)
Comment (shailesh-vaidya) [reply jenkins/performance-ci/performance-check.groovy:1]: The folder name is `performance-ci` hence used different names for files - https://github.com/shailesh-vaidya/cortx-re/tree/performance-ci-fix/jenkins/performance-ci
Also, this job is for a developer to validate performance on his custom codebase.

File: jenkins/performance-ci/nightly-performance-ci.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (gauravchaudhari02) [reply jenkins/performance-ci/nightly-performance-ci.groovy:4]: okay

File: jenkins/performance-ci/performance-check.groovy (hunk 2 above)
Comment (gauravchaudhari02) [reply jenkins/performance-ci/performance-check.groovy:1]: okay

File: scripts/performance/run-performace-tests-functions.sh (hunk 1 above)
Comment (gauravchaudhari02) [reply scripts/performance/run-performace-tests-functions.sh:99]: @shailesh-vaidya, can you answer this?

Comment (shailesh-vaidya) [reply scripts/performance/run-performace-tests-functions.sh]: Removed hardcoded values. Test job is in-progress - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Performance/job/run-performance-sanity/127/console

Comment (shailesh-vaidya) [reply scripts/performance/run-performace-tests-functions.sh]: Job is successful now. Please review

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**